jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -1687,6 +1687,131 @@ ecma_builtin_typedarray_prototype_find_index (ecma_value_t this_arg, /**< this a
   return ecma_builtin_typedarray_prototype_find_helper (this_arg, predicate, predicate_this_arg, false);
 } /* ecma_builtin_typedarray_prototype_find_index */
 
+/**
+ * The TypedArray.prototype's 'toLocaleString' single element operation routine
+ *
+ * See also:
+ *          ECMA-262 v6, 22.1.3.26 steps 7-10 and 12.b-e
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_typedarray_prototype_to_locale_string_helper (ecma_object_t *this_obj, /**< TypedArray object */
+                                                           uint32_t index) /** array index */
+{
+  lit_magic_string_id_t class_id = ecma_object_get_class_name (this_obj);
+  lit_utf8_byte_t *typedarray_buffer_p = ecma_typedarray_get_buffer (this_obj);
+
+  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+  ecma_number_t element_num = ecma_get_typedarray_element (typedarray_buffer_p + index, class_id);
+  ecma_value_t element_value = ecma_make_number_value (element_num);
+
+  if (ecma_is_value_undefined (element_value) || ecma_is_value_null (element_value))
+  {
+    ret_value = ecma_make_magic_string_value (LIT_MAGIC_STRING__EMPTY);
+  }
+  else
+  {
+    ecma_value_t next_element_obj = ecma_op_to_object (element_value);
+
+    if (ECMA_IS_VALUE_ERROR (next_element_obj))
+    {
+      return next_element_obj;
+    }
+
+    ecma_object_t *next_element_obj_p = ecma_get_object_from_value (next_element_obj);
+
+    ecma_value_t func_value = ecma_op_object_get_by_magic_id (next_element_obj_p,
+                                                              LIT_MAGIC_STRING_TO_LOCALE_STRING_UL);
+
+    if (ECMA_IS_VALUE_ERROR (func_value))
+    {
+      ecma_deref_object (next_element_obj_p);
+      return func_value;
+    }
+
+    if (ecma_op_is_callable (func_value))
```

Comment:
Is this branch covered by tests?

Reply:
this whole helper method is unnecessary since the typedArray only contains numbers, im currently working on the solution.